K2Bridge/KustoDAL/KustoDataAccess.cs

```diff
@@ -163,6 +174,66 @@ private void MapFieldCaps(IDataReader kustoResults, FieldCapabilityResponse resp
             }
         }
 
+        private async Task HandleDynamicField(FieldCapabilityResponse response, string tableName, FieldCapabilityElement fieldCapabilityElement)
+        {
+            var query = DynamicSamplePercentage.HasValue ?
+                $@"{KustoQLOperators.Let} percentage = {DynamicSamplePercentage} / 100.0;
+{KustoQLOperators.Let} table_count = {KustoQLOperators.ToScalar}({tableName} | {KustoQLOperators.Count});
+{tableName} | {KustoQLOperators.Sample} {KustoQLOperators.ToInt}({KustoQLOperators.Floor}(table_count * percentage, 1)) | {KustoQLOperators.Summarize} {KustoQLOperators.BuildSchema}({fieldCapabilityElement.Name})" :
+                $"{tableName} | {KustoQLOperators.Summarize} {KustoQLOperators.BuildSchema}({fieldCapabilityElement.Name})";
+            var (_, result) = await Kusto.ExecuteQueryAsync(new QueryData(query, tableName), RequestContext);
+            result.Read();
+            var jsonResult = result[0];
+            var stack = new Stack<(string, JObject)>();
+
+            switch (jsonResult)
+            {
+                case JArray arr:
+                    response.AddField(FieldCapabilityElementFactory.CreateFromNameAndKustoShorthandType(fieldCapabilityElement.Name, CombineValues(arr)));
+                    break;
+                case JObject obj:
+                    stack.Push((fieldCapabilityElement.Name, obj));
+                    break;
+                default:
+                    throw new InvalidOperationException($"Unexpected type {jsonResult.GetType()}");
+            }
+
+            while (stack.Count > 0)
+            {
+                var (path, jsonObject) = stack.Pop();
+                foreach (var property in jsonObject.Properties())
+                {
+                    var newPath = path + "." + property.Name;
+                    if (property.Value.Type != JTokenType.Object)
+                    {
+                        response.AddField(FieldCapabilityElementFactory.CreateFromNameAndKustoShorthandType(newPath, CombineValues(property.Value)));
+                        continue;
+                    }
+
+                    var indexer = property.Value["`indexer`"];
+                    switch (indexer)
+                    {
+                        case null:
```

Comment:
'case null' - not something you often see.

can you add documentation on what this code does

Reply:
It's part of the pattern matching thing, it's equivalent to this:

```c#
                    if (indexer == null)
                    {
                        stack.Push((newPath, (JObject)property.Value));
                    }
                    else if (indexer is JObject j)
                    {
                        stack.Push((newPath, j));
                    }
                    else
                    {
                        response.AddField(FieldCapabilityElementFactory.CreateFromNameAndKustoShorthandType(
                            newPath,
                            CombineValues(indexer)));
                    }
```
I can move the null check back if you prefer.

Anyway I added a comment, tell me WDYT

Reply:
looks better